agent/storage.go

```diff
@@ -119,6 +119,12 @@ func WalkTaskStatus(tx *bolt.Tx, fn func(id string, status *api.TaskStatus) erro
 func PutTask(tx *bolt.Tx, task *api.Task) error {
 	return withCreateTaskBucketIfNotExists(tx, task.ID, func(bkt *bolt.Bucket) error {
 		task = task.Copy()
+		oldStatus := task.Status
+
+		defer func() {
+			task.Status = oldStatus
+		}()
```

Comment:
Is this part meant to avoid the `Copy` above? If so, should that copy be removed?